runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -158,4 +167,21 @@ public void onGameStateChanged(GameStateChanged event)
 			abyssObjects.clear();
 		}
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		Query darkMageQuery = new NPCQuery().idEquals(NpcID.DARK_MAGE);
```

Comment:
You can optimize this by checking `config.hightlightDarkMage` prior to running the queries. And also by running the inventory query first and then run the NPC query only if it finds something.